### PR TITLE
code review exercise

### PR DESCRIPTION
**Pull request by hmc-cs-ctan**

### CODE REVIEW

File: HVZ/HVZ/api/views.py [hunk 1]
```diff
@@ -52,6 +59,9 @@ def form_valid(self, form):
             body = form.cleaned_data['body']
             recipient_title = form.cleaned_data['recipient']
```
Comment (hmc-cs-ctan) [inline HVZ/HVZ/api/views.py]: The commented sender "mod@claremonthvz.org" will be used for the actual mods of HvZ

File: HVZ/HVZ/api/views.py [hunk 2]
```diff
@@ -61,7 +71,7 @@ def form_valid(self, form):
         elif(recipient_title == MailerForm.ZOMBIES):
             recipients = [p.user.email for p in Player.current_players() if p.team == "Z"]
```
Comment (hmc-cs-ctan) [inline HVZ/HVZ/api/views.py]: These cases seem a bit redundant if there are more cases. An more efficient way would be better.

File: HVZ/HVZ/api/views.py [hunk 3]
```diff
@@ -61,7 +71,7 @@ def form_valid(self, form):
         elif(recipient_title == MailerForm.ZOMBIES):
             recipients = [p.user.email for p in Player.current_players() if p.team == "Z"]        
 
-        # TODO: Authentication error for sender for mod@claremonthvz.org
+        # send mail based on the given inputs
         send_mail(subject, body, sender, recipients)
```
Comment (hmc-cs-ctan) [inline HVZ/HVZ/api/views.py]: An alert of failed to send to an email address will be turned on once we have the basic functionalities.

File: HVZ/HVZ/api/views.py (hunk 2 above)
Comment (AdamSchulze) [reply HVZ/HVZ/api/views.py]: I agree. Especially if we plan to sort players in groups where there are more than two options. Perhaps we could change the list comprehension condition to read in the mailer form choice.

File: HVZ/HVZ/api/views.py [hunk 4]
```diff
@@ -34,12 +35,18 @@ def json_get_all_emails(request):
...
     success_url = '/success/'
 
     def dispatch(self, *args, **kwargs):
+        # used to display the form
         return super(Mailer, self).dispatch(*args, **kwargs)
 
     def get_success_url(self):
+        # after the mail is sent successfully, it goes to the success page
+        # At this point, it is the same as registration success page
+        # in the future, we will have more details
```
Comment (AdamSchulze) [inline HVZ/HVZ/api/views.py]: It's good that we recognized this! As for a solution, what page should we redirect to upon successfully sending an email? Should we simply return to the blank mailer form, or should we create a new success page?

File: HVZ/HVZ/api/views.py (hunk 2 above)
Comment (cvalleroy) [reply HVZ/HVZ/api/views.py]: I agree @hmc-cs-ctan and @AdamSchulze . If we look towards our stretch goals and want to separate players by Class or by College, these checks in the code will begin to be much more redundant. We could consider check-boxes for each group/subgroup to choose which groups will receive the email, however we would still need a more clean back-end implementation.

File: HVZ/HVZ/api/views.py [hunk 5]
```diff
@@ -34,12 +35,18 @@ def json_get_all_emails(request):
 class Mailer(FormView):
     form_class = MailerForm
     template_name = "api/mailer.html"
```
Comment (cvalleroy) [inline HVZ/HVZ/api/views.py]: Our link to the mailer page currently maps to api before the actual form for emails. Unless we plan to add other forms to the api folder (Ways of adding/removing players from the database and email groups or to view bounced emails), we might want to remove this subgroup.

**Pull request merged**